metadata-builders/src/main/java/com/linkedin/metadata/builders/search/MLModelIndexBuilder.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(MLModelUrn urn, CaveatsAndRecommendations caveatsAndRecommendations) {
+        return setUrnDerivedFields(urn);
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(MLModelUrn urn, EthicalConsiderations ethicalConsiderations) {
+        return setUrnDerivedFields(urn);
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(MLModelUrn urn, EvaluationData evaluationData) {
+        final MLModelDocument doc = setUrnDerivedFields(urn);
+
+        if (evaluationData.hasEvaluationData()) {
+            DatasetUrnArray datasetUrns = evaluationData.getEvaluationData()
+                .stream()
+                .map(BaseData::getDataset)
+                .collect(Collectors.toCollection(DatasetUrnArray::new));
+            doc.setEvaluationDatasets(datasetUrns);
+        }
+        return doc;
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(MLModelUrn urn, InstitutionalMemory institutionalMemory) {
+        return setUrnDerivedFields(urn);
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(@Nonnull MLModelUrn urn, @Nonnull IntendedUse intendedUse) {
+        return setUrnDerivedFields(urn);
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(@Nonnull MLModelUrn urn, @Nonnull Metrics metrics) {
+        return setUrnDerivedFields(urn);
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(@Nonnull MLModelUrn urn, @Nonnull MLModelFactorPrompts modelFactorPrompts) {
+        return setUrnDerivedFields(urn);
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(@Nonnull MLModelUrn urn, @Nonnull MLModelProperties mlModelProperties) {
+        final MLModelDocument doc = setUrnDerivedFields(urn);
+
+        if (mlModelProperties.hasDate()) {
+            doc.setCreatedTimestamp(mlModelProperties.getDate());
+        }
+
+        if (mlModelProperties.hasDescription()) {
+            doc.setDescription(mlModelProperties.getDescription());
+        }
+
+
+        return doc;
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(@Nonnull MLModelUrn urn, @Nonnull Ownership ownership) {
+        final StringArray owners = BuilderUtils.getCorpUserOwners(ownership);
+        return setUrnDerivedFields(urn)
+            .setHasOwners(!owners.isEmpty())
+            .setOwners(owners);
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(@Nonnull MLModelUrn urn, @Nonnull QuantitativeAnalyses quantitativeAnalyses) {
+        return setUrnDerivedFields(urn);
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(@Nonnull MLModelUrn urn, @Nonnull Status status) {
+        return setUrnDerivedFields(urn)
+            .setRemoved(status.isRemoved())
+            .setActive(status.isRemoved());
```

Comment:
Two fields representing the exact same logic doesn't seem ideal to me (redundant logic, wastage of storage resource, etc). Did you want to use `Deprecation` aspect to populate `active` field instead?